src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DirectoryHelper.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public static class DirectoryHelper
+    {
+        public static IEnumerable<string> GetFilteredFiles(string workspaceDirectory, string searchPattern, IReadOnlyCollection<string> ignoredNames, ILogger? logger)
```

Comment:
Also, would it be good idea to potentially null check some of these params?

Reply:
Yeah, even with nullable its a good idea to null check public methods. My question though would be does this need to be public? I assume nobody outside of the Razor solution is actually calling it, and I _think_ the standard thing in the org would be to use internal for most things, and have IVTs between our projects.

Obviously the latter being a completely separate PR 😛

Reply:
You're correct, this should be internal.